src/api_manager/context/request_context.cc

```diff
@@ -292,6 +301,64 @@ void RequestContext::FillHttpHeaders(const Response *response,
   }
 }
 
+void RequestContext::FillJwtPayloads(service_control::ReportRequestInfo *info) {
+  auto serverConfig = service_context_->config()->server_config();
+  if (serverConfig->has_service_control_config() &&
+      serverConfig->service_control_config().log_jwt_payload().size() != 0) {
+    JwtCache &jwt_cache = service_context()->jwt_cache();
+    JwtCache::ScopedLookup lookup(&jwt_cache, auth_token_);
+    // FillJwtPayloads is called when reporting to service control, which is
+    // after CheckAuth, so the userInfo should always be cached.
+    if (!lookup.Found()) {
+      return;
+    }
+    const auto &jwt_payloads =
+        serverConfig->service_control_config().log_jwt_payload();
+
+    for (const auto &payload : jwt_payloads) {
+      char *json_copy = strdup(lookup.value()->user_info.claims.c_str());
+      grpc_json *property_json =
```

Comment:
grpc_json is grpc internal lib,  we want to put all code depends on it into one place which is auth/lib.  Could you move these code into that folder.

Reply:
moved.